salt/orchestrate/aws/mitxpro_autoscaling.sls

```diff
@@ -0,0 +1,85 @@
+{% set env_settings = salt.cp.get_file_str("salt://environment_settings.yml")|load_yaml %}
+{% set ENVIRONMENT = salt.environ.get('ENVIRONMENT', 'mitxpro-production') %}
+{% set BUSINESS_UNIT = salt.environ.get('BUSINESS_UNIT', env_data.business_unit) %}
+{% set purpose = salt.grains.get('purpose', 'xpro-production') %}
+{% set env_data = env_settings.environments[ENVIRONMENT] %}
+{% set purpose_data = env_data.purposes[purpose] %}
+{% set sqs_queue = ENVIRONMENT ~ env_data.provider_services.sqs.queue %}
+{% set sns_topic = ENVIRONMENT ~ env_data.provider_services.sns.topic %}
+{% set edx_codename = purposes[PURPOSE].versions.codename %}
+
+{% set region = 'us-east-1' %}
+{% set AWS_ACCOUNT_ID = salt.vault.read('secret-operations/global/aws-account-id') %}
+{% set ami_id = salt.sdb.get('sdb://consul/edx_{}_{}_ami_id'.format(ENVIRONMENT, edx_codename)) %}
+
+create_{{ sqs_queue }}-sqs-queue:
+  boto_sqs.present:
+    - name: {{ sqs_queue }}
+    - region: {{ region }}
+    - attributes:
+        Policy: {"Version":"2012-10-17","Id":"arn:aws:sqs:{{ region }}:{{ AWS_ACCOUNT_ID }}:{{ sqs_queue }}/SQSDefaultPolicy","Statement":[{"Effect":"Allow","Principal":{"AWS":["arn:aws:iam::{{ AWS_ACCOUNT_ID }}:role/mitx-salt-master-role"]},"Action":"SQS:*","Resource":"arn:aws:sqs:{{ region }}:{{ AWS_ACCOUNT_ID }}:{{ sqs_queue }}}]}
+
+create_{{ sns_topic }}-sns-topic:
+  boto_sns.present:
+    - name: {{ sns_topic }}
+    - region: {{ region }}
+    - subscriptions:
+        - protocol: sqs
+        - endpoint: 'arn:aws:sqs:{{ region }}:{{ AWS_ACCOUNT_ID }}:{{ sqs_queue }}'
+
+create_autoscaling_group:
+  boto_asg.present:
+    - name: edx-{{ ENVIRONMENT }}-autoscaling-group
```

Comment:
Makes it sound extremely redundant having the `purpose` here as well. It's already clear from the environment name `mitxpro-production` what this is for. Just makes it sound so convoluted adding the purpose this making it `edx-mitxpro-production-xpro-production-autoscaling-group`

Reply:
Yeah, it's clear in the salt code, I'm just thinking of the case where we decide to autoscale residential and then we have a conflict in the AG name and want to prevent that situation.